### PR TITLE
parallel and sequential flows should consider IN_PROGRESS status

### PR DESCRIPTION
**Pull request by nirarg**

### CODE REVIEW

**Comment by nirarg** [comment]
This PR is following to this comment in previous PR:
https://github.com/parodos-dev/parodos/pull/184#discussion_r1153600468
@RichardW98 do you think there are other places in the code should be modified?

File: workflow-engine/src/test/java/com/redhat/parodos/workflows/workflow/SequentialFlowTest.java [hunk 1]
```diff
@@ -26,57 +26,111 @@
...
+		validateWorks(works, failureWork);
+
+		// Second work return null
+		failureWork = 1;
+		prepareWorks(works, null, failureWork);
+
+		SequentialFlow.Builder.aNewSequentialFlow().named("testFlow").execute(initialWorkUnits).then(nextWorkUnits)
+				.build().execute(workContext);
+
+		validateWorks(works, failureWork);
```
[the remark this answers is not in the document]
Comment (nirarg) [reply workflow-engine/src/test/java/com/redhat/parodos/workflows/workflow/SequentialFlowTest.java]: Changed the tests, include seperating to different test methods
Please take a look

File: workflow-engine/src/test/java/com/redhat/parodos/workflows/workflow/ParallelFlowReportTest.java [hunk 2]
```diff
@@ -35,30 +35,50 @@ public class ParallelFlowReportTest {
...
-		Assertions.assertThat(parallelFlowReport.getError()).isEqualTo(exception);
+		Assertions.assertThat(failedParallelFlowReport.getError()).isEqualTo(exception);
 	}
 
 	@Test
 	public void testGetReports() {
-		Assertions.assertThat(parallelFlowReport.getReports()).hasSize(2);
+		Assertions.assertThat(completedParallelFlowReport.getReports()).hasSize(2);
+		Assertions.assertThat(failedParallelFlowReport.getReports()).hasSize(3);
+		Assertions.assertThat(progressParallelFlowReport.getReports()).hasSize(4);
```
[the remark this answers is not in the document]
Comment (nirarg) [reply workflow-engine/src/test/java/com/redhat/parodos/workflows/workflow/ParallelFlowReportTest.java:81]: I don't see the added value for adding here any other verification, I think it enough

File: workflow-engine/src/main/java/com/redhat/parodos/workflows/workflow/ParallelFlowReport.java [hunk 3]
```diff
@@ -86,12 +86,16 @@ void addAll(List<WorkReport> workReports) {
 	 */
 	@Override
 	public WorkStatus getStatus() {
+		WorkStatus workStatus = WorkStatus.COMPLETED;
 		for (WorkReport report : reports) {
 			if (report.getStatus().equals(WorkStatus.FAILED)) {
 				return WorkStatus.FAILED;
 			}
+			if (!report.getStatus().equals(WorkStatus.COMPLETED)) {
+				workStatus = report.getStatus();
```
Comment (RichardW98) [inline workflow-engine/src/main/java/com/redhat/parodos/workflows/workflow/ParallelFlowReport.java:95]: I think the parallel workflow should be in_progress if any of its works is in_progress ? the logic here seems to set workStatus as of the last item in reports?

**Pull request merged**